### PR TITLE
Add non-fully active doc test for Geolocation

### PR DESCRIPTION
**Pull request by marcoscaceres**
Test for: 
https://github.com/w3c/geolocation-api/pull/97

### CODE REVIEW

File: geolocation-API/iframe.html [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<!DOCTYPE html>
```
Comment (reillyeon) [inline geolocation-API/iframe.html:1]: I've seen most test suites create a `resources` subdirectory for non-test files like this.

File: geolocation-API/non-fully-active.https.html [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      GeolocationPositionError.POSITION_UNAVAILABLE,
+      "getCurrentPosition returns an error on non-fully-active document"
+    );
+
+    // Re-attach, and go back to fully active.
+    document.body.appendChild(iframe);
+    iframe.contentWindow.opener = window;
+    await new Promise((r) =>
+      window.addEventListener("message", r, { once: true })
+    );
```
Comment (reillyeon) [inline geolocation-API/non-fully-active.https.html:72]:
```suggestion
    await new Promise((resolve) =>
      window.addEventListener("message", resolve, { once: true })
    );
```

Comment (reillyeon) [reply geolocation-API/non-fully-active.https.html:72]: What posts this message?

File: geolocation-API/non-fully-active.https.html [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    // And we are back to fully active.
+    let watchId;
+    let position = await new Promise((resolve, reject) => {
+      watchId = iframe.contentWindow.navigator.geolocation.watchPosition(
+        resolve,
+        reject
+      );
+    });
+    assert_true(Number.isInteger(watchId), "Expected some number for watchId");
+    ok(position, "Expected a position");
```
Comment (reillyeon) [inline geolocation-API/non-fully-active.https.html:83]: I'm not familiar with this function. Do you mean `assert_true`?

File: geolocation-API/iframe.html (hunk 1 above)
Comment (marcoscaceres) [reply geolocation-API/iframe.html:1]: Good suggestion. Will move.

File: geolocation-API/non-fully-active.https.html (hunk 2 above)
Comment (marcoscaceres) [reply geolocation-API/non-fully-active.https.html:72]: Oops! good catch. That's copy/pasta from the Gecko test.

File: geolocation-API/non-fully-active.https.html (hunk 3 above)
Comment (marcoscaceres) [reply geolocation-API/non-fully-active.https.html:83]: More copy/pasta.

File: geolocation-API/non-fully-active.https.html [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    // Create the iframe, wait for it to load...
+    const iframe = document.createElement("iframe");
+    iframe.src = "resources/iframe.html";
+    iframe.allow = "geolocation";
+    document.body.appendChild(iframe);
+
+    // ...wait for the iframe to load...
+    await new Promise((r) =>
+      iframe.addEventListener("load", r, { once: true })
+    );
```
Comment (reillyeon) [inline geolocation-API/non-fully-active.https.html:24]:
```suggestion
    await new Promise((resolve) =>
      iframe.addEventListener("load", resolve, { once: true })
    );
```

**Review by reillyeon — APPROVED** [review]

**Comment by marcoscaceres** [comment]
Thanks for all the great suggestions @reillyeon! Much improved. 

Looking forward to seeing how it's implemented in Chrome!

**Pull request merged**